File: mobile/lib/models/movie.dart
class Movie {
  final int id;
  final String title;
  final String? overview;
  final String? overviewTr;
  final String? genres;
  final String? posterUrl;
  final int? releaseYear;
  final double? letterboxdRating;
  final String? contentType;
  final String? director;
  final String? actors;
  final String? tagline;
  final String? taglineTr;
  final double? positivePct;

  Movie({
    required this.id,
    required this.title,
    this.overview,
    this.overviewTr,
    this.genres,
    this.posterUrl,
    this.releaseYear,
    this.letterboxdRating,
    this.contentType,
    this.director,
    this.actors,
    this.tagline,
    this.taglineTr,
    this.positivePct,
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      overview: json['overview'],
      overviewTr: json['overview_tr'],
      genres: json['genres'],
      posterUrl: json['poster_url'],
      releaseYear: json['release_year'],
      letterboxdRating: (json['letterboxd_rating'] as num?)?.toDouble(),
      contentType: json['content_type'],
      director: json['director'],
      actors: json['actors'],
      tagline: json['tagline'],
      taglineTr: json['tagline_tr'],
      positivePct: (json['positive_pct'] as num?)?.toDouble(),
    );
  }

  String get displayOverview => overviewTr ?? overview ?? '';
  String get displayTagline => taglineTr ?? tagline ?? '';
  String get displayGenre => genres?.split(',').first.trim() ?? '';
  double get displayRating => letterboxdRating ?? 0;
  bool get isMovie => contentType == 'Movie';
}

File: backend/models/movie.py
from sqlalchemy import Float, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from backend.core.database import Base


class Movie(Base):
    __tablename__ = "movies"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    overview: Mapped[str | None] = mapped_column(Text, nullable=True)
    overview_tr: Mapped[str | None] = mapped_column(Text, nullable=True)
    genres: Mapped[str | None] = mapped_column(String(255), nullable=True)
    themes: Mapped[str | None] = mapped_column(Text, nullable=True)
    poster_url: Mapped[str | None] = mapped_column(String(512), nullable=True)
    release_year: Mapped[int | None] = mapped_column(Integer, nullable=True)
    actors: Mapped[str | None] = mapped_column(Text, nullable=True)
    director: Mapped[str | None] = mapped_column(String(255), nullable=True)
    tagline: Mapped[str | None] = mapped_column(String(512), nullable=True)
    tagline_tr: Mapped[str | None] = mapped_column(String(512), nullable=True)
    runtime: Mapped[int | None] = mapped_column(Integer, nullable=True)
    letterboxd_rating: Mapped[float | None] = mapped_column(Float, nullable=True)
    audience_score: Mapped[float | None] = mapped_column(Float, nullable=True)
    tomato_meter: Mapped[float | None] = mapped_column(Float, nullable=True)
    total_reviews: Mapped[int | None] = mapped_column(Integer, nullable=True)
    positive_pct: Mapped[float | None] = mapped_column(Float, nullable=True)
    content_type: Mapped[str | None] = mapped_column(String(50), nullable=True, default="Movie")
    trailer_key: Mapped[str | None] = mapped_column(String(32), nullable=True)

    watch_history: Mapped[list["WatchHistory"]] = relationship(
        "WatchHistory", back_populates="movie", cascade="all, delete-orphan"
    )
    watchlist: Mapped[list["Watchlist"]] = relationship(
        "Watchlist", back_populates="movie", cascade="all, delete-orphan"
    )

